sycl/include/sycl/ext/oneapi/matrix/matrix-unified.hpp

```diff
@@ -43,14 +44,16 @@ struct joint_matrix {
   mutable sycl::ext::oneapi::detail::joint_matrix_cuda<T, Use, Rows, Cols,
                                                        Layout>
       cuda_impl;
+#elif defined(__HIP_PLATFORM_AMD_MFMA__)
+  sycl::ext::oneapi::detail::joint_matrix_hip<T, Use, Rows, Cols, Layout>
+      hip_impl;
```

Comment:
Maybe we should rename `cuda_impl` and `hip_impl` as something identical like `matrix_impl` or similar?

Reply:
@JackAKirk done in https://github.com/intel/llvm/pull/11485/commits/af5cc07e230684a6eddaf81a37c0d14d4c532c35